### PR TITLE
fix update failed when status file not exist

### PR DESCRIPTION
**Pull request by vanloswang**

### CODE REVIEW

**Comment by eddyp** [comment]
I think I have ran into the same issue when building from scratch a new rootfs, but I suspected it was due to some broken logic in the MARK_ONLY code path.

**Comment by longsleep** [comment]
This patch has at least one flaw and i also do not see why it should be required in the first place as the existing code works just fine when the $CURRENFILE does not exist. Please explain your issues so i get the idea what is the problem.

File: simpleimage/platform-scripts/pine64_update_kernel.sh [hunk 1]
```diff
@@ -48,7 +48,11 @@ if [ -e "${CURRENTFILE}" ]; then
 fi
 
 echo "Checking for update ..."
-ETAG=$(curl -f -I -H "If-None-Match: \"${CURRENT}\"" -s "${URL}"|grep ETag|awk -F'"' '{print $2}')
+if [ -z ${CURRENT} ]; then
+    ETAG=$(curl -f -I -s "${URL}" | grep etag | awk -F'"' '{print $2}')
```
Comment (longsleep) [inline simpleimage/platform-scripts/pine64_update_kernel.sh:52]: etag will not match, must be ETag

File: simpleimage/platform-scripts/pine64_update_uboot.sh [hunk 2]
```diff
@@ -32,7 +32,11 @@ if [ -e "${CURRENTFILE}" ]; then
 fi
 
 echo "Checking for update ..."
-ETAG=$(curl -f -I -H "If-None-Match: \"${CURRENT}\"" -s "${URL}"|grep ETag|awk -F'"' '{print $2}')
+if [ -z ${CURRENT} ]; then
+    ETAG=$(curl -f -I -s "${URL}" | grep etag | awk -F'"' '{print $2}')
```
Comment (longsleep) [inline simpleimage/platform-scripts/pine64_update_uboot.sh:36]: etag does not match, must be ETag

**Comment by vanloswang** [comment]
@longsleep In my country the http respones is etag, maybe we should use 'grep -E'.

**Comment by eddyp** [comment]
@vanloswang To accomodate both cases, you probably need 'grep -i' instead.

**Comment by vanloswang** [comment]
@eddyp my mistake. yes it should be 'grep -i'.

**Pull request closed without merging**